btf/types.go

```diff
@@ -568,6 +843,11 @@ func (c *cycle) copy() Type {
 	return &cpy
 }
 
+func (c *cycle) equal(other Type) bool {
+	_, ok := other.(*cycle)
+	return ok
```

Comment:
I think this should always be `false`, because `walkType` never recurses into a cycle.